packages/jest-circus/src/index.js

```diff
@@ -29,23 +31,23 @@ describe.skip = (blockName: BlockName, blockFn: BlockFn) =>
 const _dispatchDescribe = (blockFn, blockName, mode?: BlockMode) => {
   dispatch({blockName, mode, name: 'start_describe_definition'});
   blockFn();
-  dispatch({name: 'finish_describe_definition'});
+  dispatch({blockName, mode, name: 'finish_describe_definition'});
 };
 
-const _addHook = (fn: HookFn, hookType: HookType) =>
-  dispatch({fn, hookType, name: 'add_hook'});
-const beforeEach = (fn: HookFn) => _addHook(fn, 'beforeEach');
-const beforeAll = (fn: HookFn) => _addHook(fn, 'beforeAll');
-const afterEach = (fn: HookFn) => _addHook(fn, 'afterEach');
-const afterAll = (fn: HookFn) => _addHook(fn, 'afterAll');
+const _addHook = (fn: HookFn, hookType: HookType, timeout: ?number) =>
+  dispatch({fn, hookType, name: 'add_hook', timeout});
+const beforeEach: THook = (fn, timeout) => _addHook(fn, 'beforeEach', timeout);
+const beforeAll: THook = (fn, timeout) => _addHook(fn, 'beforeAll', timeout);
+const afterEach: THook = (fn, timeout) => _addHook(fn, 'afterEach', timeout);
+const afterAll: THook = (fn, timeout) => _addHook(fn, 'afterAll', timeout);
 
-const test = (testName: TestName, fn?: TestFn) =>
-  dispatch({fn, name: 'add_test', testName});
+const test = (testName: TestName, fn?: TestFn, timeout?: number) =>
```

Comment:
that's expected though! :)
in this function `timeout` is optional, and on line37, it's always passed, but can be null